Cabal/src/Distribution/Simple/Test.hs

```diff
@@ -122,9 +139,29 @@ test args pkg_descr lbi flags = do
     >>= filterM doesFileExist . map (testLogDir </>)
     >>= traverse_ removeFile
 
+  -- We configured the unit-ids of libraries we should cover in our coverage
+  -- report at configure time into the local build info. At build time, we built
+  -- the hpc artifacts into the extraCompilationArtifacts directory, which, at
+  -- install time, is copied into the ghc-pkg database files.
+  -- Now, we get the path to the HPC artifacts and exposed modules of each
+  -- library by querying the package database keyed by unit-id:
+  let coverageFor = fromFlagOrDefault [] (configCoverageFor (configFlags lbi))
+  ipkginfos <- catMaybes <$> mapM (getInstalledPackageById verbosity lbi) coverageFor
```

Comment:
Shouldn't it be an error at this stage if we can't find the .mix files for something we wanted the coverage for?

Reply:
Definitely. Doing so.